samples/cpp/speech_sample/main.cpp

```diff
@@ -993,63 +440,34 @@ int main(int argc, char* argv[]) {
-                                    MemoryBlob::CPtr moutput = as<MemoryBlob>(outputBlob);
-
-                                    if (!moutput) {
-                                        throw std::logic_error(
-                                            "We expect output to be inherited from MemoryBlob, "
-                                            "but in fact we were not able to cast output to MemoryBlob");
-                                    }
+
+                                    ov::runtime::Tensor outputBlob =
+                                        inferRequest.inferRequest.get_tensor(executableNet.outputs()[0]);
+                                    if (!FLAGS_oname.empty())
+                                        outputBlob =
+                                            inferRequest.inferRequest.get_tensor(executableNet.outputs().back());
                                     // locked memory holder should be alive all time while access to its buffer happens
-                                    auto moutputHolder = moutput->rmap();
                                     auto byteSize = numScoresPerFrame * sizeof(float);
-                                    std::memcpy(outputFrame, moutputHolder.as<const void*>(), byteSize);
+                                    std::memcpy(outputFrame, outputBlob.data<float>(), byteSize);
                                 }
                                 if (!FLAGS_r.empty()) {
                                     /** Compare output data with reference scores **/
-                                    if (!outputs.empty()) {
-                                        newOutputInfo[outputs[next_output]] = cOutputInfo[outputs[next_output]];
-                                    } else {
-                                        newOutputInfo = cOutputInfo;
-                                    }
-                                    Blob::Ptr outputBlob =
-                                        inferRequest.inferRequest.GetBlob(newOutputInfo.rbegin()->first);
-                                    MemoryBlob::CPtr moutput = as<MemoryBlob>(outputBlob);
-                                    if (!moutput) {
-                                        throw std::logic_error(
-                                            "We expect output to be inherited from MemoryBlob, "
-                                            "but in fact we were not able to cast output to MemoryBlob");
-                                    }
-                                    // locked memory holder should be alive all time while access to its buffer happens
-                                    auto moutputHolder = moutput->rmap();
+                                    ov::runtime::Tensor outputBlob =
+                                        inferRequest.inferRequest.get_tensor(executableNet.outputs()[0]);
+                                    if (!FLAGS_oname.empty())
```

Comment:
Can we use if , else block in this case?

Reply:
in this case we have common value and extra value with output's parameter name so we don't need else block